tools/go-cli/Dockerfile

```diff
@@ -32,38 +32,58 @@ RUN cd /src/github.com/go-whisk-cli && echo "Dependencies list requires restruct
 # All of the Go CLI binaries will be placed under a build folder
 RUN mkdir /src/github.com/go-whisk-cli/build
 
-# The BUILDTIME argument is used as the CLI build time property (wsk property get --all)
-# Disabled for now to speed build times
-#ARG BUILDTIME
-
-# Build the Go wsk CLI binaries
-ENV GOOS=windows
-ENV EXEC=wsk.exe
-ENV ARCHS="386 amd64"
-RUN for GOARCH in $ARCHS; do \
+# Build the Go wsk CLI binaries and compress resultant binaries
+RUN build_cli () {  \
+      echo "Building for OS '$4' and architecture '$2'" && \
+      export GOARCH=$2 && \
+      export GOOS=$4 && \
+      if [ $2 = arm ]; then \
+          export GOARM=7; \
+        fi && \
       cd /src/github.com/go-whisk-cli && \
-      go build -ldflags "-X main.CLI_BUILD_TIME=`date -u '+%Y-%m-%dT%H:%M:%S%:z'`" -v -o build/$GOOS/$GOARCH/$EXEC main.go && \
-      zip build/$GOOS/$GOARCH/openwhisk-win-$GOARCH.zip build/$GOOS/$GOARCH/$EXEC; \
-    done
+      go build -ldflags "-X main.CLI_BUILD_TIME=`date -u '+%Y-%m-%dT%H:%M:%S%:z'`" -v -o build/$1/$2/$3 main.go; \
+    }; \
 
-ENV GOOS=darwin
-ENV ARCHS="386 amd64 arm arm64"
-ENV EXEC=wsk
-RUN for GOARCH in $ARCHS; do \
-      cd /src/github.com/go-whisk-cli && \
-      go build -ldflags "-X main.CLI_BUILD_TIME=`date -u '+%Y-%m-%dT%H:%M:%S%:z'`" -v -o build/mac/$GOARCH/$EXEC main.go && \
-      tar -cf build/mac/$GOARCH/openwhisk-mac-$GOARCH.tar build/mac/$GOARCH/$EXEC && \
-      gzip < build/mac/$GOARCH/openwhisk-mac-$GOARCH.tar > build/mac/$GOARCH/openwhisk-mac-$GOARCH.tar.gz && \
-      rm -f build/mac/$GOARCH/openwhisk-mac-$GOARCH.tar; \
-    done
+    get_compressed_name() { \
+        product_name="OpenWhisk_CLI" && \
+        if [ $2 = amd64 ]; then \
+            comp_name="$product_name-$1"; \
+        elif [ $2 = 386 ]; then \
+            comp_name="$product_name-$1-32bit"; \
+        else  \
+            comp_name="$product_name-$1-$2"; \
+        fi && \
+        echo $comp_name; \
+    }; \
 
-ENV GOOS=linux
-ENV ARCHS="386 amd64 arm arm64 ppc64 ppc64le mips64 mips64le"
-ENV EXEC=wsk
-RUN for GOARCH in $ARCHS; do \
-      cd /src/github.com/go-whisk-cli && \
-      go build -ldflags "-X main.CLI_BUILD_TIME=`date -u '+%Y-%m-%dT%H:%M:%S%:z'`" -v -o build/$GOOS/$GOARCH/$EXEC main.go && \
-      tar -cf build/$GOOS/$GOARCH/openwhisk-$GOOS-$GOARCH.tar build/$GOOS/$GOARCH/$EXEC && \
-      gzip < build/$GOOS/$GOARCH/openwhisk-$GOOS-$GOARCH.tar > build/$GOOS/$GOARCH/openwhisk-$GOOS-$GOARCH.tar.gz && \
-      rm -f build/$GOOS/$GOARCH/openwhisk-$GOOS-$GOARCH.tar; \
+    compress_binaries() { \
+        comp_name=$(get_compressed_name $1 $2) && \
+        cd build/$1/$2 && \
+        if [ $4 = tgz ]; then \
+            echo "Compressing binary '$3' to '$comp_name.tgz'" && \
+            tar -cvzf $comp_name.tgz $3 && \
```

Comment:
Creating this tgz with this command will give us the auto expand like in Safari Downloads?